Sources/StreamChatUI/ChatChannelList/Tests/ChatChannelListVC_SnapshotTests.swift

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+@testable import StreamChatUI
+import XCTest
+
+class ChatChannelListVC_SnapshotTests: XCTestCase {
+    var view: ChatChannelListItemView!
+    var vc: ChatChannelListVC!
+    var mockedChannelListController: ChatChannelListController_Mock<NoExtraData>!
+    
+    override func setUp() {
+        super.setUp()
+        UIConfig.default.channelList.channelListItemSubviews.avatarView = MockAvatarView.self
+        UIConfig.default.currentUser.currentUserViewAvatarView = MockCurrentChatUserAvatarView.self
+        vc = ChatChannelListVC()
+        mockedChannelListController = ChatChannelListController_Mock.mock()
+        vc.controller = mockedChannelListController
+    }
+    
+    func test_populated() {
+        mockedChannelListController.simulate(
+            channels: [
+                .mock(
+                    cid: .init(type: .messaging, id: "test_channel1"),
+                    name: "Channel 1",
+                    lastMessageAt: .init(timeIntervalSince1970: 1_611_951_526_000)
+                ),
+                .mock(
+                    cid: .init(type: .messaging, id: "!members:test_channel2"),
+                    name: "Channel 2",
+                    lastMessageAt: .init(timeIntervalSince1970: 1_611_951_527_000),
+                    members: [.mock(id: "luke", name: "Luke Skywalker", isOnline: true)]
+                ),
+                .mock(
+                    cid: .init(type: .messaging, id: "test_channel3"),
+                    name: "Channel 3",
+                    lastMessageAt: .init(timeIntervalSince1970: 1_611_951_528_000),
+                    unreadCount: ChannelUnreadCount(messages: 4, mentionedMessages: 2),
+                    latestMessages: [
+                        ChatMessage.mock(
+                            id: "1", text: "This is a long message. How the UI will adjust?", author: .mock(id: "Vader2")
+                        )
+                    ]
+                ),
+                .mock(
+                    cid: .init(type: .messaging, id: "test_channel4"),
+                    name: "Channel 4",
+                    lastMessageAt: .init(timeIntervalSince1970: 1_611_951_529_000),
+                    latestMessages: [
+                        ChatMessage.mock(id: "2", text: "Hello", author: .mock(id: "Vader")),
+                        ChatMessage.mock(id: "1", text: "Hello2", author: .mock(id: "Vader2"))
+                    ]
+                )
+            ],
+            changes: []
+        )
+        AssertSnapshot(vc, isEmbededInNavigationController: true)
```

Comment:
Here, I tried to test every possible logic in this view controller snapshot. With this approach, we reduce the number of screenshots we need to save in comparison if we did the tests per component. So the advantage of this, is that in a single view controller test we test all the components logic and all trait variants. This results already in +/- 8 pictures, which is all the variants we currently support. If we had to do this per component, it would increase a lot the number of pictures stored.

Reply:
Great 👍

Reply:
On other hand you lose "single reason to fail" test property.
So if something goes wrong, you need to find yourself from diff which exactly state is bugged and etc.

Considering size of SDK, I'm sure amount of images will growth anyway. And for such cases Git-LFS exists 
https://git-lfs.github.com

Reply:
@ManWithBear I hear you but isn't the failure ambiguity an inherent characteristic of snapshot tests? I think about them more like integration tests than unit tests. I guess we could try to reduce this. For example in `ChannelListVC` by introducing some kind of "dummy" view instead of real channel items. But that would also mean we'd have to add more actual unit tests to test that the cells are loaded with correct data. However, we'd probably still need to have a snapshot of the "real" cell just to test that the sizing and layout work correctly as a whole.

So yeah, maybe this can be reduced by testing the different cell states separately but we will still need some "master" snapshot that will just verify everything works well together, too. IMO the approach proposed by @nuno-vieira is a good-enough start. We can always introduce more granularity if we see the lack of it causes issues.

Reply:
@ManWithBear I think that snapshot testing each component is fine for some cases (Especially for the messages cells), but for the most part, it is more realistic to test the views in a view controller, and we will find problems that we would not with just testing a view in isolation. 

Yes I'm aware of git LFS and we probably will need to use it when this scales a lot, but for now, is better if we can minimize the images we save. I also agree with @VojtaStavik that I see more advantages in treating snapshot tests as integration tests than unit tests.

It should be easy to see what went wrong, if the test fails, we can see the Diff image, so that should not be a problem:
![image](https://user-images.githubusercontent.com/12814114/106589140-9449a280-6543-11eb-83f1-9149c5a7d48b.png)